test/end-to-end/signup.test.js

```diff
@@ -342,6 +344,151 @@ describe("SuperTokens SignUp", function () {
+            formFieldErrors = await getFieldErrors(page);
+            assert.deepStrictEqual(formFieldErrors, ["Please check Terms and conditions"]);
+        });
+
+        it("Check if custom values are part of the signup payload", async function () {
+            const customFields = {
+                terms: true, // checked
+                ratings: "best",
+            };
+            const requestHandler = (request) => {
+                if (request.url().includes(SIGN_UP_API) && request.method() === "POST") {
+                    Object.keys(customFields).every((key) => {
+                        assert.strictEqual(data[key], customFields[key]);
+                    });
+                }
+                return request.continue();
+            };
+            await page.setRequestInterception(true);
+            page.on("request", requestHandler);
+
+            // Fill and submit the form with custom fields
+            try {
+                await setInputValues(page, [
+                    { name: "email", value: "john.doe@supertokens.io" },
+                    { name: "password", value: "Str0ngP@assw0rd" },
+                ]);
+
+                await setSelectDropdownValue(page, 'select[name="ratings"]', customFields["ratings"]);
+
+                // Check terms and condition checkbox
+                let termsCheckbox = await waitForSTElement(page, '[name="terms"]');
+                await page.evaluate((e) => e.click(), termsCheckbox);
+
+                await submitForm(page);
+            } finally {
+                page.off("request", requestHandler);
+                await page.setRequestInterception(false);
+            }
+        });
+    });
+
+    // Default values test
+    describe("Signup default value for fields test", function () {
+        beforeEach(async function () {
+            // set cookie and reload which loads the form fields with default values
+            await page.evaluate(() => window.localStorage.setItem("SHOW_DEFAULT_FIELDS", "YES"));
+
+            await page.reload({
+                waitUntil: "domcontentloaded",
+            });
+            await toggleSignInSignUp(page);
+        });
+
+        it("Check if default values are set already", async function () {
+            const fieldsWithDefault = {
+                country: "India",
+                ratings: "best",
+            };
+
+            // regular input field default value
+            const countryInput = await getInputField(page, "country");
+            const defaultCountry = await countryInput.evaluate((f) => f.value);
+            assert.strictEqual(defaultCountry, fieldsWithDefault["country"]);
+
+            // custom dropdown default value is also getting set correctly
+            const ratingsDropdown = await waitForSTElement(page, "select");
+            const defaultRating = await ratingsDropdown.evaluate((f) => f.value);
+            assert.strictEqual(defaultRating, fieldsWithDefault["ratings"]);
+        });
+
```

Comment:
add one more test in which we have the default values, and when you click on sign up without changing anything, those get sent to the API call.